### ISSUE TITLE
Support the opposite of a "Grace Time"

### ISSUE THREAD

**Issue by danemacmillan** [healthchecks/healthchecks#236]
What I mean by this is if we know the average run time for a job takes `n` amount of time, we are currently able to set a grace time if it takes longer than `n`, but we're not able to set the opposite of a grace time if it takes *less* than `n`.

This has happened to me a few times where the tools being used are not sending a correct exit code, and it ends up finishing very fast--abnormally fast--but because it is treated as a successful completion, the ping to healthchecks.io is able to proceed. For example, I have a job that indexes a lot of data, and usually takes about 50 minutes, but a couple weeks back it was completing in 15 to 20 minutes, but I was not alerted to the problem until someone in merchandising noticed erroneous data in our application. After reviewing the issue, this was actually happening for close to a week, and it took this long for it to be noticed.

**Comment by RyanSchw** [comment]
To go off of that, I would also like to see support of a time before the cron is supposed to be executed. I had a job that ran at 01:00 and ended up executing at 00:58 for whatever reason, but it was labeled as "Down" since completion beat the 01:00 mark.

**Comment by mgolisch** [comment]
yeah same problem here,  as a workaround we just set the shedule to be earlier than the actual cronjob is executed and adjust the graceperiod accordingly.
But some sort of feature to make that sort of thing configurable would be nicer.

**Comment by ax-meyer** [comment]
I would love to see a "minimum duration" feature as well, would be very helpful for cases where an erroneous success is send.
